trl/models/modeling_base.py

```diff
@@ -108,3 +112,69 @@ def save_pretrained(self, *args, **kwargs):
         Save the pretrained model to a directory.
         """
         return self.pretrained_model.save_pretrained(*args, **kwargs)
+
+
+def create_reference_model(model: PreTrainedModelWrapper, share_layers: int = None, pattern: str = None):
+    """
+    Creates a static reference copy of a model. Note that model will be in `.eval()` mode.
+    
+    Args:
+        model (`PreTrainedModelWrapper`): model to be copied
+        share_layers (`int`, *optional*): first layers are shared between both models and kept frozen. 
+        pattern (`str`, *optional*): The shared layers are selected with a string pattern 
+            (e.g. "transformer.h.{layer}" for GPT2) and if a custom pattern is necessary it can be passed here.
+
+    Returns
+        `PreTrainedModelWrapper`
+    """
+
+    parameter_names = [n for n, _ in model.named_parameters()]
+    ref_model = deepcopy(model)
+
+    # if no layers are shared, return copy of model
+    if share_layers is None:
```

Comment:
do we need to extend this logic to handle the case when `share_layers=0` or does the indexing start with `0`?

Reply:
indeed indexing starts at 0 so this should be fine